### ISSUE TITLE
Unhelpful error for MVC controller taking constructor params

### ISSUE THREAD

**Issue by jtlapp** [jooby-project/jooby#3541]
I spent the past few hours tracking down the cause of the error, "Router not found: `app.controllers.HomeController`. Make sure Jooby annotation processor is configured properly."

Annotations were configured properly, and the problem was not in `HomeController`. Rather, I get this error attempting to do manual DI on a controller. Jooby MVC doesn't seem to like controllers taking arguments.

It would really have helped for the error to have said something that led me in that direction. It would also have helped for the MVC docs to have told me of this constraint.

Here's the code that caused the above error:

```java
package com.jtlapp.jvmvsjs.joobymvc.controllers;

import io.jooby.annotation.GET;
import io.jooby.annotation.Path;
import io.jooby.annotation.PathParam;

import java.util.concurrent.CompletableFuture;
import java.util.concurrent.ScheduledExecutorService;
import java.util.concurrent.TimeUnit;

@Path("/api")
public class ApiController {

    private final ScheduledExecutorService scheduler;

    public ApiController(ScheduledExecutorService scheduler) {
        this.scheduler = scheduler;
    }

    @GET
    @Path("/sleep/{millis}")
    public CompletableFuture<String> sleep(@PathParam int millis) {
        var future = new CompletableFuture<String>();

        scheduler.schedule(() -> {
            future.complete("");
        }, millis, TimeUnit.MILLISECONDS);

        return future;
    }
}
```

The error goes away when I remove the constructor's parameter (and the code that depends on this parameter).

Of course, it's always possible that my correction addressed a symptom rather than the cause. However, the `HomeController` is servicing requests after removing the constructor parameter, so we can be sure that the error message is wrong.

**Comment by jknack** [comment]
Hi,

 You are doing:

```
  mvc(ApiController.class);
```

right?

If that is the case, Jooby is expecting to get the controller using:

- Default/empty constructor
- Asking to DI (service registry) to get the instance

In your case none of those will work. Also, this method relay on kind of reflection which we change in the new jooby-apt.

For now just replace that line with:

```
  mvc(new ApiController_(scheduler));
```

The generated class: `ApiController_` mimics/copy the controller constructors.

I will work on improve the error message.

**Comment by jtlapp** [comment]
Your response clued me in to the problem: I was calling `mvc(new ApiController`(scheduler))` rather than `mvc(new ApiController_(scheduler))`. I was missing the underscore. After adding this, I was able to build and run from the command line.

The IntilliJ linter doesn't like that underscore until after I build, so I guess I need to install an extension.

Thanks again for your help! It would definitely have done me some good to emphasize that underscore in the text. I did notice it, but I assumed it was a typo because IntelliJ didn't like it.
